public/assets/scripts/blocklyflasher.js

```diff
@@ -0,0 +1,1790 @@
+        {
+            document.getElementById('plugin0').flashBootloader(
+                (this.portslist.selectedIndex == -1 || programmer['communication'] != 'serial')?'':this.portslist.options[this.portslist.selectedIndex].text,
+                programmer['protocol'],
+                programmer['communication'],
+                programmer['speed'],
+                programmer['force'],
+                programmer['delay'],
+                board['bootloader']['high_fuses'],
+                board['bootloader']['low_fuses'],
+                (typeof board['bootloader']['extended_fuses'] === "undefined")?'':board['bootloader']['extended_fuses'],
+                (typeof board['bootloader']['unlock_bits'] === "undefined")?'':board['bootloader']['unlock_bits'],
+                (typeof board['bootloader']['lock_bits'] === "undefined")?'':board['bootloader']['lock_bits'],
+                board['build']['mcu'],
+                bootloader_callback
+            );
+        };
+
+        this.clickedPort = function()
+        {
+            var port = $("#cb_cf_ports").val();
+            var actionId = 43;
+            var metaData = {
+                "selectedPort": port,
+                "tabID": this.tabID
+            };
+            createLogFlasher(actionId, metaData);
+        };
+
+        this.oldPort = null;
+
+        this.logPorts = function () {
+            var oldPort;
+            if (this.oldPort == null)
+                oldPort = '';
+            else
+                oldPort = this.oldPort;
+
+            var newPort;
+            if (this.currentPorts.length > 0)
+                newPort = $("#cb_cf_ports option:selected").text();
+            else
+                newPort = '';
+
+            this.oldPort = newPort;
+
+            if (oldPort != newPort) {
+                var actionId = 38;
+                var metaData = {
+                    "oldPort": oldPort,
+                    "newPort": newPort,
+                    "tabID": this.tabID
+                };
+                createLogFlasher(actionId, metaData);
+            }
+        };
+
+        this.savePort = function() {
+            var cb = this;
+            if(typeof Lawnchair !== "undefined")
+            {
+                new Lawnchair(function () {
+                    if(cb.currentPorts.length > 0) {
+                        var port = $("#cb_cf_ports option:selected").text();
+                        this.save({key:'port', name:port});
+                    }
+                });
+            }
+            cb.logPorts();
+        };
+
+        this.loadPort = function() {
+            var cb = this;
+            if (this.currentPorts.length > 0) {
+                $('#cb_cf_ports').find('option[value=""]').remove();
+                if(typeof Lawnchair !== "undefined") {
+                    Lawnchair(function () {
+                        this.exists('port', function (exists) {
+                            if (exists) {
```

Comment:
By this line I'm already looking for paper to note what each indent is and what if/elses do. 

There is a school that says to
1. Not have more than one indent level 
2. Never use else

Both can work 99% of the time and once you adopt them code looks much nicer and cleaner. Look up the subject of code calisthenics, they come from OOP but can be applied to most languages for most part.